kore/src/Kore/Log/DecidePredicateUnknown.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE NoStrict #-}
+{-# LANGUAGE NoStrictData #-}
+
+{- |
+Copyright   : (c) Runtime Verification, 2020-2021
+License     : BSD-3-Clause
+-}
+module Kore.Log.DecidePredicateUnknown (
+    OnDecidePredicateUnknown (..),
+    DecidePredicateUnknown (..),
+    throwDecidePredicateUnknown,
+) where
+
+import Control.Exception (
+    Exception (..),
+    throw,
+ )
+import Data.Limit (Limit (..))
+import Debug
+import Kore.Attribute.SourceLocation (
+    SourceLocation (..),
+ )
+import Kore.Internal.Predicate (
+    Predicate,
+ )
+import Kore.Internal.Predicate qualified as Predicate
+import Kore.Internal.Variable
+import Log
+import Prelude.Kore
+import Pretty (
+    Pretty (..),
+ )
+import Pretty qualified
+import SMT qualified
+
+data OnDecidePredicateUnknown
+    = WarnSimplificationEquationInApplication (Maybe SourceLocation)
+    | ErrorInApplication (Maybe SourceLocation)
+    | ErrorInMatchWith
+    | ErrorInSimplifyClaimRule
+    | ErrorInFilterMultiOr
+    deriving stock (Show, Eq)
```

Comment:
A new ADT which keeps track of the action ot take and where the warning/error roughly originated in the codebase.